lib/parser.js

```diff
@@ -206,11 +210,29 @@ async function customDocumentOperations(parsedJSON, asyncapiYAMLorJSON, initialF
   await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
 }
 
-async function validateAndConvertMessage(msg, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload) {
+async function validateAndConvertMessage(msg, channelName ,originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload) {
   //check if the message has been parsed before
   if (xParserMessageParsed in msg && msg[String(xParserMessageParsed)] === true) return;
   const defaultSchemaFormat = getDefaultSchemaFormat(parsedAsyncAPIDocument.asyncapi);
   const schemaFormat = msg.schemaFormat || defaultSchemaFormat;
+  const parser = PARSERS[String(schemaFormat)];
+  const errorMap = new Map();
+  errorMap.set(`${tilde(channelName)}/${Operation.name}/${msg}/${schemaFormat}`, schemaFormat);
```

Comment:
`validateAndConvertMessage` is called twice, because in AsyncAPI document, message can be in 2 locations:
- under channel/operation/message -> in this case function is called in an `Object.entries` iteration where you have access to `opName`, so you can actually pass it to `validateAndConvertMessage` instead of doing `Operation.name`
- under components/messages -> in this case there is no operation and error location will not be `${tilde(channelName)}/${Operation.name}/${msg}/${schemaFormat}` but under components like `messages/${msgName}/${schemaFormat}`

so yeah, something I did not explain before is that you need to figure out first if you are dealing with a message on `channels` or `components`. So I would definitely not add `channelName` argument to `validateAndConvertMessage` in the middle of the list of arguments but at the end as option maybe, like that it can be either `{ channelName: "myName" }` or `{ component: true }` and then you can easily check what option is passed and what to do in the code.

What do you think?